fn.go

```diff
@@ -137,11 +140,26 @@ func (f *Function) RunFunction(ctx context.Context, req *fnv1.RunFunctionRequest
 	// TODO(negz): Where the heck is the newline at the end of this key
 	// coming from? Bug in crossplane render?
 	key := strings.Trim(string(b), "\n")
+
+	// Extract optional base URL from credentials
+	var baseURL string
+	if baseURLBytes, ok := c.Data[credBaseURLKey]; ok {
+		baseURL = strings.Trim(string(baseURLBytes), "\n")
+	}
+
+	// Extract optional model from credentials, default to gpt-4
+	model := defaultModel
+	if modelBytes, ok := c.Data[credModelKey]; ok {
+		model = strings.Trim(string(modelBytes), "\n")
+	}
```

Comment:
I think this is generally fine for now. One thing that I'm not sure about is if these configurations should stay in the secret, as roughly a configuration singleton, or if we should move them to the input. Both options are dynamic for the lifespan of the function request, but the model and the base url aren't really "sensitive". Also moving it to the input may make it a little more apparent how the function is configured within the `Operation` pipeline itself.

Just some things to think about as we try out this new functionality 👍